index.js

```diff
@@ -3,7 +3,7 @@
 var BufferStreams = require('bufferstreams');
 var htmlmin = require('html-minifier');
 var gutil = require('gulp-util');
-var objectAssign = require('object-assign');
```

Comment:
I'm unwilling to change this.

`gulp` itself depends on `object-assign` (`gulp` -> `vinyl-fs` -> `object-assign`), so even if users install `gulp-htmlmin`, npm doesn't download additional `object-assign` thanks to npm's [package deduplication](http://blog.npmjs.org/post/109921886740/a-belated-npm-weekly-3).

Switching to `extend-shallow` requires the gulp-htmlmin users to install the additonal package because all the upstream projects don't use  `extend-shallow`.